bundles/org.openhab.core.model.sitemap/src/org/openhab/core/model/sitemap/validation/SitemapValidator.xtend

```diff
@@ -102,4 +107,14 @@ class SitemapValidator extends AbstractSitemapValidator {
                 SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT__MIN_VALUE));
         }
     }
+    
+    @Check
+    def void checkInputHintParameter(Input i) {
+        if (i.inputHint !== null && !ALLOWED_HINTS.contains(i.inputHint)) {
+            val node = NodeModelUtils.getNode(i)
+            val line = node.getStartLine()
+            error("Input on item '" + i.item + "' has invalid inputHint '" + i.inputHint + "' at line " + line,
```

Comment:
Did you test what happens if the item is not defined ?

Reply:
It will not even run this code and fail earlier, refusing to load the sitemap. That's because it is already an error in the grammar xtext.

Reply:
Sorry, I did not see that you made the item mandatory.